app/models/thread.php

```diff
@@ -0,0 +1,66 @@
+<?php
+class Thread extends AppModel
+{
+    public $validation = array(
+        'title' => array(
+            'length' => array(
+                'validate_between', 1, 30,
+            ),
+        ),
+    );
+
+    public static function getAll()
+    {
+        $threads = array();
+        $db = DB::conn();
+        $rows = $db->rows('SELECT * FROM thread');
+        foreach ($rows as $row){
+            $threads[] = new Thread($row);
+        }
+        return $threads;
+    }
+
+    public static function get($id)
+    {
+        $db = DB::conn();
+        $row = $db->row('SELECT * FROM thread WHERE id = ?', array($id));
+        return new self($row);
+    }
+
+    public function getComments()
+    {
+        $comments = array();
+        $db = DB::conn();
+        $rows = $db->rows('SELECT * FROM comment WHERE thread_id = ? ORDER BY created ASC', array($this->id));
+        foreach ($rows as $row)
+            {
+                $comments[] = new Comment($row);
+            }
+        return $comments;
+    }
+
+    public function write(Comment $comment)
+    {
+        $db = DB::conn();
+        $db->query(
+            'INSERT INTO comment SET thread_id = ?, username = ?, body = ?, created = NOW()', array($this->id, $comment->username, $comment->body));
```

Comment:
Also, please transfer this procedure in comment model as well since it involves the comment table. You can modify this method like this
```php
public function writeComment($body)
{
    Comment::create($body, $this);
}
```
then in `models/comment.php`
```php
public static function Comment($body, Thread $thread)
{
    $db = DB::conn();
    $params = array(
        'thread_id' => $thread->id,
        'username' => $thread->username,
        'body' => $body,
        'created' => date('Y-m-d H:i:s')
    );
    $db->insert('comment', $params);
}